exchanges/zb/zb.go

```diff
@@ -331,17 +334,18 @@ func (z *ZB) SendAuthenticatedHTTPRequest(ep exchange.URL, httpMethod string, pa
 	// Expiry of timestamp doesn't appear to be documented, so making a reasonable assumption
 	ctx, cancel := context.WithDeadline(context.Background(), now.Add(15*time.Second))
 	defer cancel()
-	err = z.SendPayload(ctx, &request.Item{
+
+	item := &request.Item{
 		Method:        httpMethod,
 		Path:          urlPath,
-		Body:          strings.NewReader(""),
 		Result:        &intermediary,
 		AuthRequest:   true,
 		Verbose:       z.Verbose,
 		HTTPDebugging: z.HTTPDebugging,
 		HTTPRecording: z.HTTPRecording,
-		Endpoint:      f,
-	})
+	}
+
+	err = z.SendPayload(ctx, f, func() (*request.Item, error) { return item, nil })
 	if err != nil {
 		return err
```

Comment:
This function also uses timestamp based signature/headers. I would advise inlining. ZB has rate limits